framework/CodeInterfaces/MELCOR/melcorInterface.py

```diff
@@ -55,38 +51,34 @@ def generateCommand(self, inputFiles, executable, clargs=None, fargs=None):
       @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
     """
-    subReturnCommand = self.melgenApp.generateCommand(inputFiles, executable, clargs, fargs)
+    subReturnCommand = MelgenApp.generateCommand(self,inputFiles, executable, clargs, fargs)
     outputfile = 'GEN_OUTPUT'
     returnCommand = subReturnCommand[0],outputfile
     return returnCommand
-
-  def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
+  
+  def checkForOutputFailure(self,output,workingDir):
     """
-      This generates a new input file depending on which sampler has been chosen
-      @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
-      @ In, oriInputFiles, list, list of the original input files
-      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
-      @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
-             where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
-      @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
+      This method is called by the RAVEN code at the end of each run  if the return code is == 0.
+      This method needs to be implemented by the codes that, if the run fails, return a return code that is 0
+      This can happen in those codes that record the failure of the job (e.g. not converged, etc.) as normal termination (returncode == 0)
+      This method can be used, for example, to parse the outputfile looking for a special keyword that testifies that a particular job got failed
+      (e.g. in RELAP5 would be the keyword "********")
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, failure, bool, True if the job is failed, False otherwise
     """
-    if "dynamicevent" in samplerType.lower():
-      raise IOError("Dynamic Event Tree-based samplers not implemented for MELCOR yet!")
-    indexes  = []
-    inFiles  = []
-    origFiles= []
-    #FIXME possible danger here from reading binary files
-    for index,inputFile in enumerate(currentInputFiles):
-      if inputFile.getExt() in self.getInputExtension():
-        indexes.append(index)
-        inFiles.append(inputFile)
-    for index,inputFile in enumerate(origInputFiles):
-      if inputFile.getExt() in self.getInputExtension():
-        origFiles.append(inputFile)
-    parser = GenericParser.GenericParser(inFiles)
-    parser.modifyInternalDictionary(**Kwargs)
-    parser.writeNewInput(currentInputFiles,origFiles)
-    return currentInputFiles
+    failure = False
```

Comment:
This is an important method... in this way it is completely bypassed... Isn't a string/sentence/whatever that MELCOR writes somewhere that tells us that no crash/no convergence issue/ no problem occurred?